src/com/facebook/buck/jvm/scala/ScalaLibraryDescription.java

```diff
@@ -0,0 +1,170 @@
+
+    DefaultJavaLibrary defaultJavaLibrary =
+        resolver.addToIndex(
+            new DefaultJavaLibrary(
+                params.appendExtraDeps(
+                    Iterables.concat(
+                        BuildRules.getExportedRules(
+                            Iterables.concat(
+                                params.getDeclaredDeps().get(),
+                                resolver.getAllRules(args.providedDeps.get()))),
+                        scalac.getDeps(pathResolver))),
+                pathResolver,
+                args.srcs.get(),
+                validateResources(
+                    pathResolver,
+                    params.getProjectFilesystem(),
+                    args.resources.get()),
+                /* generatedSourceFolder */ Optional.<Path>absent(),
+                /* proguardConfig */ Optional.<SourcePath>absent(),
+                /* postprocessClassesCommands */ ImmutableList.<String>of(),
+                params.getDeclaredDeps().get(),
+                resolver.getAllRules(args.providedDeps.get()),
+                new BuildTargetSourcePath(abiJarTarget),
+                /* additionalClasspathEntries */ ImmutableSet.<Path>of(),
+                new ScalacToJarStepFactory(
+                    scalaBuckConfig.getScalac(resolver),
+                    ImmutableList.<String>builder()
+                        .addAll(scalaBuckConfig.getCompilerFlags())
+                        .addAll(args.extraArguments.get())
+                        .build()),
+                args.resourcesRoot,
+                args.mavenCoords,
+                args.tests.get()));
+
+    resolver.addToIndex(
+        CalculateAbi.of(
+            abiJarTarget,
+            pathResolver,
+            params,
+            new BuildTargetSourcePath(defaultJavaLibrary.getBuildTarget())));
+
+    return defaultJavaLibrary;
+  }
+
+  @Override
+  public Iterable<BuildTarget> findDepsForTargetFromConstructorArgs(
+      BuildTarget buildTarget,
+      Function<Optional<String>, Path> cellRoots,
+      Arg constructorArg) {
+    return ImmutableList.<BuildTarget>builder()
+        .add(scalaBuckConfig.getScalaLibraryTarget())
+        .addAll(scalaBuckConfig.getScalacTarget().asSet())
+        .build();
+  }
+
+  @SuppressFieldNotInitialized
+  public static class Arg {
+    public Optional<ImmutableSortedSet<SourcePath>> srcs;
+    public Optional<ImmutableSortedSet<SourcePath>> resources;
+    public Optional<ImmutableList<String>> extraArguments;
+    // Note: scala does not have a exported_deps because scala needs the transitive closure of
+    // dependencies to compile. deps is effectively exported_deps.
+    public Optional<ImmutableSortedSet<BuildTarget>> providedDeps;
+    public Optional<ImmutableSortedSet<BuildTarget>> deps;
+
+    @Hint(isInput = false)
+    public Optional<Path> resourcesRoot;
+    public Optional<String> mavenCoords;
+
+    @Hint(isDep = false) public Optional<ImmutableSortedSet<BuildTarget>> tests;
```

Comment:
For consistency, either keep the @Hint annotation in the same line or breakline like in line #163.